packages/hawtio/src/plugins/camel/routes-service.ts

```diff
@@ -205,6 +256,102 @@ class RoutesService {
   async deleteRoute(objName: string) {
     await jolokiaService.execute(objName, 'remove()')
   }
+
+  async dumbRoutesStatsXML(routesNode: MBeanNode | null): Promise<string | null> {
+    if (!routesNode) {
+      throw new Error('No route/routes node provided')
+    }
+
+    let xml = null
+    let mbeanName = routesNode.getProperty(contextNodeType)
+
+    try {
```

Comment:
Put the logic inside named functions instead of using a comment imo

Reply:
@joshiraez Thanks for the suggestion, but I don't think I understand. The comment here is just for explanation why I'm checking mBeanName.

Reply:
I feel like it's a bit hard to understand. I was thinking about putting them in their own functions to avoid the comment.

I have a small question though, you execute with an empty string for the mbeanName in case there is no object name for route node? Or you just catch the error?

In any case, personally I'd try to make clearer the logic for it as it's really hard to understand the if else. The operations are very similar and the intention is not clear. What do you think about something like this?

```
const routeNodeOperation = dumpRouteStatsAsXml
const routesFolderOperation = dumpRoutesStatsAsXml

const mbeanName = routesNode.getProperty(contextNodeType)

const operationForMBean = mbeanName ? routesFolderOperation : routeNodeOperation
const mbeanToQuery = mbeanName ? mbeanName :  routesNode.objectName

try {
  const xml = await jolokiaService.execute(mbeanToQuery ?? '', operationForMBean, [true, true])
} catch (error) {
  throw new Error('Failed to dump routes stats from mbean: ' + mbeanName + error)
}

if (!xml) {
    throw new Error('Failed to extract any xml from mbean: ' + mbeanName)
}

return xml as string
```

Reply:
Using the ternary in mbeanToQuery because the ?? operator only works for void types. This way I preserve the original logic, you don't need to add a comment as it can be seen that it depends on if mbean is set or not, avoid the if-else and avoid the use of let. What do you think?

![image](https://user-images.githubusercontent.com/17336236/231678054-44892b89-8707-4c0e-9370-8ea5bc35e547.png)

Reply:
(If you don't like it we can leave it as is, it's just that I found it a bit confusing at first discerning the logic with the comment)